File: 99_archive/03_component_md/ref./claude-plugins-develop/plugins/cds-ds/samples/bottom-popup/atoms/top-navigation/ios/WDSTopNavigation.swift
// WDSTopNavigation.swift
// Chord Design System — Top Navigation Atom

import SwiftUI

// MARK: - Enums

public enum WDSTopNavigationLeft {
    case none
    case back
    case custom
}

// MARK: - Token Colors

private extension Color {
    static let wdsTextDefault = Color.black
    static let wdsSurface     = Color.white
}

// MARK: - WDSTopNavigation

public struct WDSTopNavigation<LeftContent: View>: View {
    let title: String
    let showClose: Bool
    let leftType: WDSTopNavigationLeft
    let leftContent: () -> LeftContent
    let onClose: () -> Void
    let onBack: () -> Void

    public init(
        title: String,
        showClose: Bool = true,
        leftType: WDSTopNavigationLeft = .none,
        onClose: @escaping () -> Void = {},
        onBack: @escaping () -> Void = {},
        @ViewBuilder leftContent: @escaping () -> LeftContent = { EmptyView() as! LeftContent }
    ) {
        self.title = title
        self.showClose = showClose
        self.leftType = leftType
        self.onClose = onClose
        self.onBack = onBack
        self.leftContent = leftContent
    }

    public var body: some View {
        HStack(spacing: 0) {
            // Left slot — 공간 유지로 타이틀 중앙 정렬 방지 (Figma: 타이틀 좌측 정렬)
            Group {
                switch leftType {
                case .none:
                    Color.clear
                        .frame(width: 40, height: 40)
                case .back:
                    Button(action: onBack) {
                        Image(systemName: "chevron.left")
                            .font(.system(size: 16, weight: .medium))
                            .foregroundColor(.wdsTextDefault)
                            .frame(width: 40, height: 40)
                    }
                    .buttonStyle(.plain)
                case .custom:
                    leftContent()
                        .frame(width: 40, height: 40)
                }
            }

            // Title — 좌측 정렬
            Text(title)
                .font(.system(size: 16, weight: .bold))
                .foregroundColor(.wdsTextDefault)
                .lineLimit(1)
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.horizontal, 4)

            // Close button
            if showClose {
                Button(action: onClose) {
                    Image(systemName: "xmark")
                        .font(.system(size: 13, weight: .medium))
                        .foregroundColor(.wdsTextDefault)
                        .frame(width: 40, height: 40)
                }
                .buttonStyle(.plain)
                .accessibilityLabel("닫기")
            } else {
                Color.clear
                    .frame(width: 40, height: 40)
            }
        }
        .frame(height: 48)
        .padding(.horizontal, 4)
        .background(Color.wdsSurface)
    }
}

// MARK: - Convenience (no custom left)

public extension WDSTopNavigation where LeftContent == EmptyView {
    init(
        title: String,
        showClose: Bool = true,
        leftType: WDSTopNavigationLeft = .none,
        onClose: @escaping () -> Void = {},
        onBack: @escaping () -> Void = {}
    ) {
        self.title = title
        self.showClose = showClose
        self.leftType = leftType
        self.onClose = onClose
        self.onBack = onBack
        self.leftContent = { EmptyView() }
    }
}

// MARK: - Preview

#if DEBUG
struct WDSTopNavigation_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: 0) {
            Divider()
            Text("Default — title + close").font(.caption).foregroundColor(.secondary).frame(maxWidth: .infinity, alignment: .leading).padding(.horizontal, 16).padding(.top, 12)
            WDSTopNavigation(title: "Title", showClose: true)
            Divider()

            Text("Back + title + close").font(.caption).foregroundColor(.secondary).frame(maxWidth: .infinity, alignment: .leading).padding(.horizontal, 16).padding(.top, 12)
            WDSTopNavigation(title: "Title", showClose: true, leftType: .back)
            Divider()

            Text("showClose=false").font(.caption).foregroundColor(.secondary).frame(maxWidth: .infinity, alignment: .leading).padding(.horizontal, 16).padding(.top, 12)
            WDSTopNavigation(title: "Title", showClose: false)
            Divider()
        }
    }
}
#endif

File: 99_archive/03_component_md/ref./claude-plugins-develop/plugins/cds-ds/samples/bottom-popup/atoms/top-navigation/android/WDSTopNavigation.kt
// WDSTopNavigation.kt
// Chord Design System — Top Navigation Atom

package com.weverse.ds.component

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// ── Enums ─────────────────────────────────────────────────────────────────────
enum class WDSTopNavigationLeft { None, Back, Custom }

// ── Token Colors ──────────────────────────────────────────────────────────────
private val WdsColorTextDefault = Color(0xFF000000)
private val WdsColorSurface     = Color(0xFFFFFFFF)

// ── WDSTopNavigation ──────────────────────────────────────────────────────────
@Composable
fun WDSTopNavigation(
    modifier: Modifier = Modifier,
    title: String,
    showClose: Boolean = true,
    leftType: WDSTopNavigationLeft = WDSTopNavigationLeft.None,
    onClose: () -> Unit = {},
    onBack: () -> Unit = {},
    customLeft: (@Composable () -> Unit)? = null
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .height(48.dp)
            .background(WdsColorSurface)
            .padding(horizontal = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Left slot — 빈 자리도 40dp 유지하여 title 좌측 정렬 보장
        Box(modifier = Modifier.size(40.dp), contentAlignment = Alignment.Center) {
            when (leftType) {
                WDSTopNavigationLeft.Back -> {
                    IconButton(
                        onClick = onBack,
                        modifier = Modifier.size(40.dp).semantics { contentDescription = "뒤로가기" }
                    ) {
                        Text("‹", fontSize = 22.sp, color = WdsColorTextDefault, fontWeight = FontWeight.Light)
                    }
                }
                WDSTopNavigationLeft.Custom -> {
                    customLeft?.invoke()
                }
                WDSTopNavigationLeft.None -> { /* empty spacer */ }
            }
        }

        // Title — 좌측 정렬 (BottomPopup 명세)
        Text(
            text = title,
            modifier = Modifier
                .weight(1f)
                .padding(horizontal = 4.dp),
            fontSize = 16.sp,
            fontWeight = FontWeight.Bold,
            color = WdsColorTextDefault,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis
        )

        // Close button
        Box(modifier = Modifier.size(40.dp), contentAlignment = Alignment.Center) {
            if (showClose) {
                IconButton(
                    onClick = onClose,
                    modifier = Modifier.size(40.dp).semantics { contentDescription = "닫기" }
                ) {
                    Text(
                        text = "×",
                        fontSize = 22.sp,
                        color = WdsColorTextDefault,
                        fontWeight = FontWeight.Light
                    )
                }
            }
        }
    }
}
